### PR TITLE
Add selection for podman/docker

### PR DESCRIPTION
**Pull request by Niklas-Koch**
### Description

This PR gives you the option to choose between podman and docker in the docker.sh script.
I came up with this idea because Red Hat and the most Red Hat-Distribution does not allow docker anymore. Instead they use podman. 
Now you can choose between both options while entering a -p for podman or a -d for docker after the ./docker.sh command.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### How Has This Been Tested?

I tested it on our testenvironment in my company.
I also tested both options, including the case that you do not choose any option.

### CODE REVIEW

**Comment by weizhouapache** [comment]
@Niklas-Koch 
for backwards compatibility, can we use docker by default ?

**Comment by svenvogel** [comment]
> @Niklas-Koch
> for backwards compatibility, can we use docker by default ?

**Comment by Niklas-Koch** [comment]
@weizhouapache 
Good idea, 
So I changed the file, that docker is the default value now, to hold the backwards compatibility. 
Now you have the possibility to use the script without any options and it will use docker as standard. But you also can you the script with an option like -d for docker (it will use docker the same way as without using the option) or -p for podman. So we have the opportunity to use podman. That means we can use the podman, which is getting more and more used and at the same time, we maintained the backwards compatibility.
I think it is now looking good to me.
Can you look through it?

File: ui/docker.sh [hunk 1]
```diff
@@ -29,5 +29,20 @@ DATE="$(date --iso-8601=seconds)"
...
+while getopts $OPTIONS OPTION
+do
+        case "$OPTION" in
+                p)USE_PODMAN=1;;
+                d)USE_DOCKER=1;;
+                *)echo "Please choose between -p for podman or -d for docker.";;
+        esac
+done
+if [[ $USE_PODMAN -eq 1 ]]; then
+        podman build -t cloudstack-ui ${LABEL_DATE} ${LABEL_GIT_REV} ${LABEL_GIT_TAG}
```
Comment (j-porsche) [inline ui/docker.sh:44]: @Niklas-Koch You missed the point at the end of the line 47
```suggestion
        podman build -t cloudstack-ui ${LABEL_DATE} ${LABEL_GIT_REV} ${LABEL_GIT_TAG} .
```

File: ui/docker.sh [hunk 2]
```diff
@@ -29,5 +29,20 @@ DATE="$(date --iso-8601=seconds)"
...
+        exit
+fi
+docker build -t cloudstack-ui ${LABEL_DATE} ${LABEL_GIT_REV} ${LABEL_GIT_TAG}
```
Comment (j-porsche) [inline ui/docker.sh:47]: @Niklas-Koch You missed the point at the end of the line 47

```suggestion
docker build -t cloudstack-ui ${LABEL_DATE} ${LABEL_GIT_REV} ${LABEL_GIT_TAG} .
```

**Comment by DaanHoogland** [comment]
@Niklas-Koch @j-porsche closing this as #5217 supercedes it. please re-open if I'm mistaking

**Pull request closed without merging**